lib/src/chrome.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:coverage/src/util.dart';
+import 'package:source_maps/parser.dart';
+
+/// Returns a Dart based hit-map containing coverage report for the provided
+/// Chrome [preciseCoverage].
+///
+/// [sourceProvider] returns the source content for the Chrome scriptId, or null
+/// if not available.
+///
+/// [sourceMapProvider] returns the associated source map content for the Chrome
+/// scriptId, or null if not available.
+///
+/// [sourceUriProvider] returns the uri for the provided sourceUrl and
+/// associated scriptId.
+Future<Map<String, dynamic>> parseChromeCoverage(
+  List<Map<String, dynamic>> preciseCoverage,
+  Future<String> Function(String scriptId) sourceProvider,
+  Future<String> Function(String scriptId) sourceMapProvider,
+  Future<Uri> Function(String sourceUrl, String scriptId) sourceUriProvider,
+) async {
+  final coverageReport = <Uri, Map<int, bool>>{};
+  for (Map<String, dynamic> entry in preciseCoverage) {
+    final String scriptId = entry['scriptId'];
+
+    final mapResponse = await sourceMapProvider(scriptId);
+    if (mapResponse == null) continue;
```

Comment:
[nit] Should we call out this behavior of ignoring values that don't get a source map or source file in the doc comment?

Reply:
Done.